### PR TITLE
[1.10] Track containers in metrics isolator.

### PR DESCRIPTION
**Pull request by rukletsov**
To prevent removal of unknown containers to metrics isolator and hence the
annoying log line, keep track of all container IDs that are passed over into
`ContainerAssigner`.

Related ticket: [DCOS_OSS-4630](https://jira.mesosphere.com/browse/DCOS_OSS-4630).

## Testing done

This particular PR has not been tested, however a similar change to the 1.11
branch has been, see https://github.com/dcos/dcos-metrics/pull/172.

### CODE REVIEW

File: mesos_module/isolator_module.cpp [hunk 1]
```diff
@@ -81,12 +91,22 @@ namespace metrics {
 
     process::Future<Nothing> cleanup(
         const mesos::ContainerID& container_id) {
-      container_assigner->unregister_container(container_id);
+      // If we haven't registered before and are not emitting metrics, e.g.,
+      // because we are a nested container in the `DEBUG` class, we have
+      // nothing to cleanup.
+      if (registered.count(container_id.value())) {
```
Comment (tillt) [inline mesos_module/isolator_module.cpp]: See master PR.

Comment (rukletsov) [reply mesos_module/isolator_module.cpp]: Updated.

**Comment by vinodkone** [comment]
Is this ready to be merged? I see failed builds.

**Comment by philipnrmn** [comment]
@vinodkone it is not ready to be merged

**Comment by philipnrmn** [comment]
I added some changes to 1.10.x which fix the build; I will rebase and merge.

**Comment by philipnrmn** [comment]
Here's the passing build for 1.10.x, post merge:
https://jenkins.mesosphere.com/service/jenkins/job/public-dcos-cluster-ops/job/dcos-metrics/job/dcos-metrics-branch/15/

**Comment by philipnrmn** [comment]
NB if you intend to bump the dcos-metrics reference downstream in DC/OS, please also cherry-pick this commit to update the build:
https://github.com/dcos/dcos/commit/6cbe1d62195a09a9f18263cc1a1b91dfd668dfe5

**Pull request merged**